### PR TITLE
Create URLDownloaderPython.py

### PR DESCRIPTION
**Pull request by jgstew**
Initial commit of new processor, originally from: https://github.com/jgstew/jgstew-recipes/blob/main/SharedProcessors/URLDownloaderPython.py

Resolves: https://github.com/autopkg/autopkg/issues/734

Because this is a new core processor, it won't be used by anyone from the start, so the risk is low.

### CODE REVIEW

**Comment by jgstew** [comment]
I created a copy of a recipe and hard coded this new processor to test it: https://github.com/jgstew/jgstew-recipes/blob/main/BigFix/FixletDebugger-test.download.recipe.yaml

I also tested it with an example that uses `prefetch_filename`: https://github.com/jgstew/jgstew-recipes/blob/main/Microsoft/SQLServerManagementStudio-Win.download.recipe.yaml

As well as an example that does not work with `etag` and `last-modified` normally: https://github.com/jgstew/jgstew-recipes/blob/main/GoToMeeting/GoToMeeting-Win.download.recipe.yaml

**Comment by jgstew** [comment]
@nmcspadden  any reason to not merge this?

File: Code/autopkglib/URLDownloaderPython.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+import os
+import ssl
+
+from hashlib import sha1, sha256, md5
+
+try:
+    from urllib.request import urlopen  # Python 3
+except ImportError:
+    print("ERROR: Not tested with Python 2")
+    from urllib2 import urlopen  # Python 2
```
Comment (nmcspadden) [inline Code/autopkglib/URLDownloaderPython.py:28]: I don't intend to maintain python 2 compatibility, so I would instead remove this entirely.

**Comment by nmcspadden** [comment]
This looks overall fine, but please run it through the precommit hooks to be sure - run it through black, isort, and flake8. 

I'm not fond of the pylint messages, since pylint is not used in this project, and I'd prefer you remove all of them.

**Comment by jgstew** [comment]
Thanks for the updates. I'll get to this soon.

**Comment by jgstew** [comment]
@nmcspadden 

- [ ] Done "run it through black, isort, and flake8"
- [ ] Done "since pylint is not used in this project, and I'd prefer you remove all of them"

**Review by homebysix — APPROVED** [review]

**Pull request merged**